### PR TITLE
TST: update cooodinates accuracy test data

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description

This is a continuation of #16976 with the addition of a commit that updates test data following a run of the `generate_ref_ast.py` script.
Let's see how this does in CI.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by eerovaher** [comment]
I didn't go over all the values, but from the quick glance I took I got the impression that the numbers aren't different at all, the updated script just writes out more digits. If it is indeed the case that limiting the number of digits in the output replicates the data as it is then the only changes would be in the file headers, and those are small enough that we can easily handle them in #16976 and this pull request wouldn't be needed.

**Comment by neutrinoceros** [comment]
Does the unified IO interface allows to set the output precision somehow ?

**Comment by eerovaher** [comment]
[See the example with the `formats` argument in our documentation.](https://astropy-astrofrog.readthedocs.io/en/latest/io/unified.html#id1)

**Comment by pllim** [comment]
If you can get the diff smaller with the precision thingy, that would be nice. My two cents, FWIW. Thanks, all!

**Comment by eerovaher** [comment]
It looks to me like the original format is 12 digits in total (meaning 9-11 digits to the right of the decimal point depending on how many digits there are to the left of it), but without trailing zeros. If there is a way to force the new data to be like that then it should be possible to make the diff much smaller.

File: astropy/coordinates/tests/accuracy/generate_ref_ast.py [hunk 1]
```diff
@@ -125,12 +148,15 @@ def ref_fk4_no_e_fk5(fnout="fk4_no_e_fk5.csv"):
...
     t.add_column(Column(name="ra_fk4", data=ra_fk4))
     t.add_column(Column(name="dec_fk4", data=dec_fk4))
-    f = open(os.path.join("data", fnout), "wb")
-    f.write(
-        f"# This file was generated with the {os.path.basename(__file__)} script, and"
-        " the reference values were computed using AST\n"
-    )
-    t.write(f, format="ascii", delimiter=",")
+    formats = {
+        k: "%0.10f" for k in "ra_in,dec_in,ra_fk5,dec_fk5,ra_fk4,dec_fk4".split(",")
```
Comment (eerovaher) [inline astropy/coordinates/tests/accuracy/generate_ref_ast.py:152]:
```suggestion
        k: "%0.10f" for k in "ra_in,dec_in,ra_fk5,dec_fk5,ra_fk4,dec_fk4".split(",")
```
```suggestion
        k: ".12g" for k in "ra_in,dec_in,ra_fk5,dec_fk5,ra_fk4,dec_fk4".split(",")
```

Comment (neutrinoceros) [reply astropy/coordinates/tests/accuracy/generate_ref_ast.py:152]: that's it ! thanks a *LOT*. There's no difference after that, so I'll close this PR and include your suggestion in #16976

**Pull request closed without merging**